### PR TITLE
Migrate ListenerTimeouts away from name-based executor

### PR DESCRIPTION
**Pull request by DaveCTurner**
Replaces the executor name with a proper executor.

Relates #99027 and friends

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java [hunk 1]
```diff
@@ -595,7 +596,7 @@ void openSession(
                 threadPool,
                 responseListener,
                 ccrSettings.getRecoveryActionTimeout(),
-                ThreadPool.Names.GENERIC,
+                chunkResponseExecutor,
```
Comment (ldematte) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java:599]: Just double checking: in current code `chunkResponseExecutor` is `threadPool.generic()`, so nothing changes. But do we want the executor we pass to `RestoreSession` to use for timeouts to be always the same as `chunkResponseExecutor` (so this code is fine) or do we want to be always `threadPool.generic()`? I'm thinking, what happens if in the future we change `chunkResponseExecutor` to be something else? Are we still good?

Comment (DaveCTurner) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java]: That's a good question. My intention here was to handle the timeout on the same threadpool as the non-timeout response. But you're right, the response is not a _chunk_ response so we shouldn't be using the chunk response executor here. In fact we handle the non-timeout response on the CCR pool, so I added a comment to suggest we come back and move this to the CCR pool too.

**Review by ldematte — APPROVED** [review]
Thanks for the clarification, LGTM!

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-2 (unrelated test failure, issue incoming)

**Pull request merged**